### PR TITLE
Fix arity check for keyword arguments

### LINKED ISSUE #9: Cannot spy on methods with named parameters
## Repro Steps
```
require 'spy'

class Foo
  def do_stuff(a, b: nil)
    puts("hello")
  end
end

f = Foo.new

Spy.on(f, :do_stuff)

f.do_stuff('val1', b: 'val2')
```

## Expected Results
no error, can investigate call arguments (including a hash of named arguments)

## Actual Results
error
```
/Users/wes/.rvm/gems/ruby-2.3.3/gems/spy-0.4.5/lib/spy/subroutine.rb:266:in `check_arity!': wrong number of arguments (2 for 1) (ArgumentError)
	from /Users/wes/.rvm/gems/ruby-2.3.3/gems/spy-0.4.5/lib/spy/subroutine.rb:204:in `invoke'
	from /Users/wes/.rvm/gems/ruby-2.3.3/gems/spy-0.4.5/lib/spy/subroutine.rb:231:in `block in override_method'
	from test.rb:13:in `<main>'
```

### PR DESCRIPTION
**Pull request by dylanahsmith**
Fixes #9

Previously keyword arguments were ignored, which prevented them from being passed in. This at least allows them to be passed in by treating them as a single optional or required argument. However, the code doesn't yet check that a hash is passed in for the keyword arguments or that the hash has the expected keys.

### CODE REVIEW

File: test/support/features.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Features
+  extend self
+
+  def keyword_args?
+    Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.0.0")
+  end
+
+  def required_keyword_args?
```
Comment (ryanong) [inline test/support/features.rb]: I would be willing to drop ruby 2.0.0 support to reduce code complexity

**Comment by ryanong** [comment]
looks like there are some failures on jruby.

If you don't want to fix those that is fine I'll try to fix it

**Comment by dylanahsmith** [comment]
For some reason I am not able to reproduce the jruby failures locally.  However, it looks like the failures isn't happening after rebasing on top of https://github.com/ryanong/spy/pull/11

**Comment by dylanahsmith** [comment]
ruby 2.0 support has been dropped to simplify the code and tests are passing.  Are there any other concerns?

**Comment by ryanong** [comment]
Thanks for the contribution I'll make a push now or when I get home

**Comment by ryanong** [comment]
@dylanahsmith pushed!

**Pull request merged**